bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/MagentaTVHandler.java

```diff
@@ -0,0 +1,575 @@
+        updateState(CHANNEL_PROG_TEXT, new StringType(""));
+        updateState(CHANNEL_PROG_START, new StringType(""));
+        updateState(CHANNEL_PROG_DURATION, new StringType(""));
+
+        updateState(CHANNEL_CHANNEL, new DecimalType(0));
+    }
+
+    private String fixEventJson(String jsonEvent) {
+        // MR401: channel_num is a string -> ok
+        // MR201: channel_num is an int -> fix JSON formatting to String
+        if (jsonEvent.contains(MR_EVENT_CHAN_TAG) && !jsonEvent.contains(MR_EVENT_CHAN_TAG + "\"")) {
+            // hack: reformat the JSON string to make it compatible with the GSON parsing
+            logger.trace("malformed JSON->fix channel_num");
+            String start = StringUtils.substringBefore(jsonEvent, MR_EVENT_CHAN_TAG); // up to "channel_num":
+            String end = StringUtils.substringAfter(jsonEvent, MR_EVENT_CHAN_TAG); // behind "channel_num":
+            String chan = StringUtils.substringBetween(jsonEvent, MR_EVENT_CHAN_TAG, ",");
+            chan = StringUtils.trim(chan);
+            return start + "\"channel_num\":" + "\"" + chan + "\"" + end;
+        }
+        return jsonEvent;
+    }
+
+    private boolean isOnline() {
+        return this.getThing().getStatus() == ThingStatus.ONLINE;
+    }
+
+    /**
+     * Check device status, if pairing
+     * events (if callbackUrl is configured)
+     */
+    @SuppressWarnings("null")
+    private void renewEventSubscription() {
+        if (control == null) {
+            return;
+        }
+        logger.debug("Check receiver status, current state  {0}/{1}",
+                this.getThing().getStatusInfo().getStatus().toString(),
+                this.getThing().getStatusInfo().getStatusDetail());
+
+        try {
+            // when pairing is completed re-new event channel subscription
+            if ((this.getThing().getStatus() != ThingStatus.OFFLINE) && !thingConfig.getVerificationCode().isEmpty()) {
+                logger.debug("Renew MR event subscription for device {0}", deviceName());
+                control.subscribeEventChannel();
+            }
+        } catch (MagentaTVException | RuntimeException e) {
+            logger.exception(e, "Re-new event subscription  for device {0} failed!", deviceName());
+        }
+        // another try: if the above SUBSCRIBE fails, try a re-connect immediatly
+
+        try {
+            if ((this.getThing().getStatusInfo().getStatusDetail() == ThingStatusDetail.COMMUNICATION_ERROR)
+                    && !thingConfig.getUserID().isEmpty()) {
+                // if we have no userID the OAuth is not completed or pairing process got stuck
+                logger.info("Reconnect media receiver {0}", deviceName());
+                connectReceiver(); // throws MagentaTVException on error
+            }
+        } catch (MagentaTVException | RuntimeException e) {
+            logger.exception(e, "Re-Connect for device {0} failed", deviceName());
```

Comment:
Reduce to DEBUG

Reply:
that is a strange situation, so it makes sense to inform the user

Reply:
Would be better to set Thing status offline and add this information to status